### PR TITLE
PPTP-1164 - GRS - Registered Society integration

### PR DESCRIPTION
**Pull request by SteveSugden**
Enable "Registered Society" GRS journey following spike 

Requires - https://github.com/hmrc/plastic-packaging-tax-registration/pull/69

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed
 - [ ] Required Environment Config has been amended/added
 - [x] Links to dependencies have been included (BE/FE work)
 - [x] User Acceptance Tests (UAT) were run locally and have passed.
 - [x] No Accessibility errors/warnings reported by Wave

image

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/OrganisationDetailsTypeController.scala [hunk 1]
```diff
@@ -110,17 +116,19 @@ class OrganisationDetailsTypeController @Inject() (
...
-                                appConfig.externalSignOutLink
-      )
+  private def incorpEntityGrsCreateRequest(implicit request: Request[_]) =
+    IncorpEntityGrsCreateRequest(appConfig.grsCallbackUrl,
+                                 Some(request2Messages(request)("service.name")),
+                                 appConfig.serviceIdentifier,
+                                 appConfig.externalSignOutLink
     )
 
+  private def getUkCompanyRedirectUr()(implicit request: Request[_]): Future[String] =
```
Comment (kiranprakashdigital) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/OrganisationDetailsTypeController.scala:126]: Typo should Url :)

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/OrganisationDetailsTypeController.scala [hunk 2]
```diff
@@ -110,17 +116,19 @@ class OrganisationDetailsTypeController @Inject() (
...
+    ukCompanyGrsConnector.createJourney(incorpEntityGrsCreateRequest)
+
+  private def getRegisteredSocietyRedirectUr()(implicit request: Request[_]): Future[String] =
```
Comment (kiranprakashdigital) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/OrganisationDetailsTypeController.scala:129]: Typo

Comment (SteveSugden) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/OrganisationDetailsTypeController.scala:129]: You must have excellent eyesight and a decent monitor.  Took me a while to spot even with you calling it out.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/forms/LiabilityWeight.scala [hunk 3]
```diff
@@ -32,7 +32,7 @@ object LiabilityWeight {
   val weightOutOfRangeError = "liabilityWeight.outOfRange.error"
 
   private val isWithinRange: Option[Long] => Boolean = weight =>
-    minTotalKg to maxTotalKg contains weight.getOrElse(minTotalKg)
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/forms/LiabilityWeight.scala:35]: This change was it improve performance BTW.   Before the change I'd noticed that 'LiabilityWeightTest' was "stuttering" for a few seconds when you ran `sbt test`

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/OrganisationDetailsTypeController.scala (hunk 2 above)
Comment (SteveSugden) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/OrganisationDetailsTypeController.scala]: Fixed

**Pull request merged**